packages/connect-core/src/utils/misc.ts

```diff
@@ -1,3 +1,54 @@
 export function toArrayEntry<T>(value: T | T[]): T[] {
   return Array.isArray(value) ? value : [value]
 }
+
+type CacheStoreCallback<T> = () => Promise<T>
+type CacheStoreCached<T> = (
+  id: string,
+  callback: CacheStoreCallback<T>
+) => Promise<T>
+
+export function createCacheStore<T = string>(
+  limit: number = 10
+): {
+  cachedIndex: (id: string) => number
+  clear: () => void
+  get: (id: string, callback: CacheStoreCallback<T>) => Promise<T>
+  touch: (index: number) => void
+} {
+  const cachedEntries: [string, T][] = Array(limit)
+
+  return {
+    clear(): void {
+      cachedEntries.length = 0
+    },
+    cachedIndex(id: string): number {
+      return cachedEntries.findIndex((entry) => entry && id === entry[0])
```

Comment:
I wonder if it's a good opportunity to prevent using `&&` and instead use `??`

Reply:
Yes! We could even check for `undefined` to be super specific since it’s what we want to test here.

Reply:
Thinking more about it, the value can only be either an array or `undefined`, so it shouldn’t be a problem. Still changing it to `??` as it is the new `||` :smile: